### PR TITLE
Fixes #21018 - Fix Rails 5 test failures

### PR DESCRIPTION
**Pull request by jturel**
On this PR all of the tests are passing on both rails 4 and 5, but only on top of two existing PRs which form the basis of my branch.

To test, simply add `:rails: '5.0'` in settings.yaml followed by a `bundle update` to check the tests against Rails 5.

### CODE REVIEW

**Comment by jlsherrill** [comment]
LGTM, mind rebasing and squashing?

**Comment by mmoll** [comment]
I think GH-6967 should go in first, apart from the rest

**Comment by jlsherrill** [comment]
aaaand its in :)

**Comment by mmoll** [comment]
💚

**Comment by jturel** [comment]
@jlsherrill rebased!

**Comment by jlsherrill** [comment]
ACK thanks @jturel !

File: app/models/katello/concerns/smart_proxy_extensions.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ module SmartProxyExtensions
                  :source     => :lifecycle_environment
 
         has_many :content_facets, :class_name => "::Katello::Host::ContentFacet", :foreign_key => :content_source_id,
-                                  :inverse_of => :content_source
+                                  :inverse_of => :content_source, :dependent => :nullify
```
Comment (jlsherrill) [inline app/models/katello/concerns/smart_proxy_extensions.rb]: i made  a note here to look into this more:  http://projects.theforeman.org/issues/21021

**Pull request merged**